### PR TITLE
Move callables another time in exception_wrapper

### PR DESCRIPTION
**Pull request by benlandrum**
The `exception_wrapper` moves callables around in `with_exception,` except for one place. This forces callables to be copyable. Use `std::move` in the remaining place.

### CODE REVIEW

File: folly/test/ExceptionWrapperTest.cpp [hunk 1]
```diff
@@ -151,6 +151,18 @@ TEST(ExceptionWrapper, with_exception_test) {
...
+  // Just needs to compile.
+  struct MoveOnly {
+    MoveOnly() = default;
+    MoveOnly(const MoveOnly&) = delete;
+    MoveOnly& operator=(const MoveOnly&) = delete;
+    MoveOnly(MoveOnly&&) noexcept = default;
+    void Foo() const {}
+  };
+  MoveOnly move_only;
+  cew.with_exception([move_only = std::move(move_only)](const std::exception&) { move_only.Foo(); };
```
Comment (yfeldblum) [inline folly/test/ExceptionWrapperTest.cpp:165]: I think we can get away with choosing an existing move-only type, such as `std::unique_ptr<int>`.

```
  // a move-only callback type:
  cew.with_exception([v = std::make_unique<int>(7)](const std::exception&) {});
```

**Review by yfeldblum — COMMENTED** [review]
Thanks! One request, listed in the code.

**Comment by benlandrum** [comment]
Hi, @yfeldblum. Any other changes requested? Could you please estimate when you'd be able to merge?

**Comment by benlandrum** [comment]
Thank you, @yfeldblum.

**Pull request closed without merging**